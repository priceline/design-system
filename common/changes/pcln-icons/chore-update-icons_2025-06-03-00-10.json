{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Update to styled-system v5 on pcln-icons, move to dev dependency and peer dependency. Use Node v20 for the build process.",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}