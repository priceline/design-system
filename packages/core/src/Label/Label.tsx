import React from 'react'
import styled from 'styled-components'
import {
  FontSizeProps,
  FontWeightProps,
  LineHeightProps,
  SpaceProps,
  TextStyleProps,
  WidthProps,
  compose,
  fontSize,
  fontWeight,
  letterSpacing,
  lineHeight,
  space,
  textStyle,
  width,
} from 'styled-system'
import { typographyAttrs } from '../utils/attrs/typographyAttrs'
import { applyVariations, getPaletteColor } from '../utils/utils'

const nowrap = (props) =>
  props.nowrap
    ? {
        whiteSpace: 'nowrap',
      }
    : null

const accessiblyHide = (props) =>
  props.hidden
    ? {
        position: 'absolute',
        width: '1px',
        height: '1px',
        clip: 'rect(1px, 1px, 1px, 1px)',
      }
    : null

const truncate = (props) =>
  props.truncate
    ? {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      }
    : null

/**
 * @public
 */
export type LabelProps = SpaceProps &
  FontSizeProps &
  FontWeightProps &
  LineHeightProps &
  TextStyleProps &
  WidthProps &
  Partial<Omit<HTMLLabelElement, 'children'>> & {
    children?: React.ReactNode
    color?: string
    autoHide?: boolean
    nowrap?: boolean
    for?: string
    as?: unknown
    truncate?: boolean
    onClick?: (evt: unknown) => void
  }

/**
 * @public
 */
export const Label: React.FC<LabelProps> & { isLabel?: boolean } = styled.label.attrs((props) => ({
  ...typographyAttrs(props),
  ...props,
}))`
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: ${getPaletteColor('base')};
  ${(props) => (props.bg ? `background-color: ${getPaletteColor(props.bg, 'base')(props)};` : '')}
  ${(props) => (props.onClick ? 'cursor: pointer;' : '')}

  ${applyVariations('Label')}

  ${nowrap}
  ${accessiblyHide}
  ${truncate}

  ${(props) => compose(fontSize, fontWeight, lineHeight, letterSpacing, space, textStyle, width)(props)}
`

Label.defaultProps = {
  color: 'text.light',
  textStyle: 'label',
}

Label.displayName = 'Label'
Label.isLabel = true
