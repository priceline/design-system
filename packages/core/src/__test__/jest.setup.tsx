import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
