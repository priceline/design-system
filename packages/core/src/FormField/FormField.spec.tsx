import { Email as EmailIcon } from 'pcln-icons'
import { render, screen } from '../__test__/testing-library'
import { Input } from '../Input/Input'
import { Label } from '../Label/Label'
import { Select } from '../Select/Select'
import { FormField } from './FormField'

// afterEach(() => {
//   // bust cache for propTypes
//   FormField.displayName = 'FormField' + Math.random()
// })

describe('FormField', () => {
  test('renders', () => {
    const { asFragment } = render(
      <FormField>
        <Label>Email Address</Label>
        <Input id='email' name='email' />
      </FormField>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with Icon', () => {
    const { asFragment } = render(
      <FormField>
        <Label>Email Address</Label>
        <EmailIcon />
        <Input id='email' name='email' />
      </FormField>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with Select ', () => {
    const { asFragment } = render(
      <FormField>
        <Label>Pick Option</Label>
        <Select />
      </FormField>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with Select and Icon', () => {
    const { asFragment } = render(
      <FormField>
        <Label>Pick Email Address</Label>
        <EmailIcon />
        <Select />
      </FormField>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with Label - forces truncate', () => {
    render(
      <FormField>
        <Label>Pick Email Address</Label>
        <Input />
      </FormField>
    )

    expect(screen.getByText('Pick Email Address')).toHaveStyleRule('white-space', 'nowrap')
    expect(screen.getByText('Pick Email Address')).toHaveStyleRule('overflow', 'hidden')
    expect(screen.getByText('Pick Email Address')).toHaveStyleRule('text-overflow', 'ellipsis')
  })

  test('renders with Label autoHide prop', () => {
    const { asFragment } = render(
      <FormField>
        <Label autoHide>Email</Label>
        <EmailIcon />
        <Input id='email' name='email' />
      </FormField>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('shows Label with autoHide prop and field value', () => {
    render(
      <FormField data-testid='form-field'>
        <Label autoHide>Email</Label>
        <EmailIcon />
        <Input id='email' name='email' value='hello@example.com' />
      </FormField>
    )
    const formField = screen.getByTestId('form-field')
    const label = formField.children[0] as HTMLElement
    expect(label).toHaveStyle('opacity: 1')
  })

  test('adds htmlFor prop to Label', () => {
    render(
      <FormField>
        <Label data-testid='label' />
        <Input id='hello' name='hello' />
      </FormField>
    )
    const label = screen.getByTestId('label')
    expect(label).toHaveAttribute('for', 'hello')
  })

  describe('disabled state', () => {
    it('renders input with icon - disabled', () => {
      const { asFragment } = render(
        <FormField disabled>
          <Label>Email Address</Label>
          <EmailIcon data-testid='email-icon' />
          <Input id='email' name='email' placeholder='Email' />
        </FormField>
      )

      expect(asFragment()).toMatchSnapshot()
    })

    it('renders select with icon - disabled', () => {
      const { asFragment } = render(
        <FormField disabled>
          <Label>Pick Email Address</Label>
          <EmailIcon />
          <Select defaultValue='Premium Economy'>
            <option>Premium Economy</option>
          </Select>
        </FormField>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
