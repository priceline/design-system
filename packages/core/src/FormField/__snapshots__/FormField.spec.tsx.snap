// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField disabled state renders input with icon - disabled 1`] = `
<DocumentFragment>
  .c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-left: 8px;
  margin-right: 8px;
  width: 24px;
}

.c8 {
  outline: none;
}

.c9 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c0cad5;
  font-size: 16px;
  border-radius: 12px;
  padding-left: 40px;
}

.c9::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c9::-moz-placeholder {
  color: #4f6f8f;
}

.c9:-ms-input-placeholder {
  color: #4f6f8f;
}

.c9::placeholder {
  color: #4f6f8f;
}

.c9::-ms-clear {
  display: none;
}

.c9:disabled {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c9:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

.c3 {
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: #4f6f8f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 10px;
  margin-left: 40px;
  padding-top: 6px;
  margin-bottom: -20px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c1 {
  border-radius: 12px;
}

.c5 {
  margin-right: -40px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  background-color: #edf0f3;
}

.c2.c2.c2 * {
  background-color: transparent;
  color: #4f6f8f;
  cursor: not-allowed;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c9 {
    font-size: 14px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
      disabled=""
    >
      <label
        class="c3"
        color="text.light"
        font-size="10"
        font-weight="700"
        for="email"
        letter-spacing="0.2px"
        style="height: 20px; transition-property: opacity; transition-duration: .1s; opacity: 1; pointer-events: none; position: relative; width: calc(100% - 40px);"
      >
        Email Address
      </label>
      <div
        class="c4"
      >
        <div
          class="c5 c6"
          data-testid="icon-field-prefix-icons"
        >
          <svg
            aria-hidden="true"
            class="c7 c8"
            data-testid="email-icon"
            fill="currentcolor"
            focusable="false"
            height="24"
            role="img"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </div>
        <input
          class="c9"
          disabled=""
          font-size="2,,1"
          id="email"
          name="email"
          placeholder="Email"
          style="transition-property: padding-top, padding-bottom; transition-duration: .1s; padding-top: 20px; padding-bottom: 8px;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FormField disabled state renders select with icon - disabled 1`] = `
<DocumentFragment>
  .c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-left: 8px;
  margin-right: 8px;
  width: 24px;
}

.c11 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  color: #4f6f8f;
  margin-left: -32px;
  width: 24px;
}

.c12 {
  outline: none;
}

.c8 {
  outline: none;
}

.c3 {
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: #4f6f8f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 10px;
  margin-left: 40px;
  padding-top: 6px;
  margin-bottom: -20px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c1 {
  border-radius: 12px;
}

.c5 {
  margin-right: -40px;
}

.c9 {
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 {
  pointer-events: none;
}

.c10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  border-style: solid;
  border-width: 1px;
  border-radius: 12px;
  color: inherit;
  display: block;
  font-family: inherit;
  text-overflow: ellipsis;
  width: 100%;
  padding: 14px 32px 14px 12px;
  border-color: #c0cad5;
  padding-left: 40px;
  font-size: 16px;
  margin: 0px;
}

.c10::-ms-expand {
  display: none;
}

.c10:hover {
  cursor: pointer;
}

.c10:disabled {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
  opacity: 1;
}

.c10:disabled ~ svg {
  color: #4f6f8f;
}

.c10:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  background-color: #edf0f3;
}

.c2.c2.c2 * {
  background-color: transparent;
  color: #4f6f8f;
  cursor: not-allowed;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c10 {
    font-size: 14px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
      disabled=""
    >
      <label
        class="c3"
        color="text.light"
        font-size="10"
        font-weight="700"
        letter-spacing="0.2px"
        style="height: 20px; transition-property: opacity; transition-duration: .1s; opacity: 1; pointer-events: none; position: relative; width: calc(100% - 40px);"
      >
        Pick Email Address
      </label>
      <div
        class="c4"
      >
        <div
          class="c5 c6"
          data-testid="icon-field-prefix-icons"
        >
          <svg
            aria-hidden="true"
            class="c7 c8"
            fill="currentcolor"
            focusable="false"
            height="24"
            role="img"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </div>
        <div
          class="c9 c4"
          width="1"
        >
          <select
            class="c10"
            disabled=""
            font-size="2,,1"
            style="transition-property: padding-top, padding-bottom; transition-duration: .1s; padding-top: 20px; padding-bottom: 8px;"
          >
            <option
              selected=""
            >
              Premium Economy
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="c11 c12 c13"
            color="text.light"
            fill="currentcolor"
            focusable="false"
            height="24"
            role="img"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FormField renders 1`] = `
.c3 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c0cad5;
  font-size: 16px;
  border-radius: 12px;
}

.c3::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c3::-moz-placeholder {
  color: #4f6f8f;
}

.c3:-ms-input-placeholder {
  color: #4f6f8f;
}

.c3::placeholder {
  color: #4f6f8f;
}

.c3::-ms-clear {
  display: none;
}

.c3:disabled {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c3:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

.c1 {
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: #4f6f8f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 10px;
  margin-left: 12px;
  padding-top: 6px;
  margin-bottom: -20px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c0 {
  border-radius: 12px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c3 {
    font-size: 14px;
  }
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="text.light"
    fontSize={10}
    fontWeight={700}
    htmlFor="email"
    letterSpacing="0.2px"
    style={
      Object {
        "height": 20,
        "opacity": 1,
        "pointerEvents": "none",
        "position": "relative",
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
        "width": "calc(100% - 12px)",
      }
    }
  >
    Email Address
  </label>
  <div
    className="c2"
  >
    <input
      className="c3"
      fontSize={
        Array [
          2,
          null,
          1,
        ]
      }
      id="email"
      name="email"
      size="lg"
      style={
        Object {
          "paddingBottom": "8px",
          "paddingTop": "20px",
          "transitionDuration": ".1s",
          "transitionProperty": "padding-top, padding-bottom",
        }
      }
    />
  </div>
</div>
`;

exports[`FormField renders with Icon 1`] = `
.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-left: 8px;
  margin-right: 8px;
  width: 24px;
}

.c6 {
  outline: none;
}

.c7 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c0cad5;
  font-size: 16px;
  border-radius: 12px;
  padding-left: 40px;
}

.c7::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c7::-moz-placeholder {
  color: #4f6f8f;
}

.c7:-ms-input-placeholder {
  color: #4f6f8f;
}

.c7::placeholder {
  color: #4f6f8f;
}

.c7::-ms-clear {
  display: none;
}

.c7:disabled {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c7:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

.c1 {
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: #4f6f8f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 10px;
  margin-left: 40px;
  padding-top: 6px;
  margin-bottom: -20px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c0 {
  border-radius: 12px;
}

.c3 {
  margin-right: -40px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c7 {
    font-size: 14px;
  }
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="text.light"
    fontSize={10}
    fontWeight={700}
    htmlFor="email"
    letterSpacing="0.2px"
    style={
      Object {
        "height": 20,
        "opacity": 1,
        "pointerEvents": "none",
        "position": "relative",
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
        "width": "calc(100% - 40px)",
      }
    }
  >
    Email Address
  </label>
  <div
    className="c2"
  >
    <div
      className="c3 c4"
      data-testid="icon-field-prefix-icons"
    >
      <svg
        aria-hidden={true}
        className="c5 c6"
        fill="currentcolor"
        focusable={false}
        height={24}
        role="img"
        tabIndex={-1}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
        />
      </svg>
    </div>
    <input
      className="c7"
      fontSize={
        Array [
          2,
          null,
          1,
        ]
      }
      id="email"
      name="email"
      size="lg"
      style={
        Object {
          "paddingBottom": "8px",
          "paddingTop": "20px",
          "transitionDuration": ".1s",
          "transitionProperty": "padding-top, padding-bottom",
        }
      }
    />
  </div>
</div>
`;

exports[`FormField renders with Label autoHide prop 1`] = `
.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-left: 8px;
  margin-right: 8px;
  width: 24px;
}

.c6 {
  outline: none;
}

.c7 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c0cad5;
  font-size: 16px;
  border-radius: 12px;
  padding-left: 40px;
}

.c7::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c7::-moz-placeholder {
  color: #4f6f8f;
}

.c7:-ms-input-placeholder {
  color: #4f6f8f;
}

.c7::placeholder {
  color: #4f6f8f;
}

.c7::-ms-clear {
  display: none;
}

.c7:disabled {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c7:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

.c1 {
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: #4f6f8f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 10px;
  margin-left: 40px;
  padding-top: 6px;
  margin-bottom: -20px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c0 {
  border-radius: 12px;
}

.c3 {
  margin-right: -40px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c7 {
    font-size: 14px;
  }
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="text.light"
    fontSize={10}
    fontWeight={700}
    htmlFor="email"
    letterSpacing="0.2px"
    style={
      Object {
        "height": 20,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "relative",
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
        "width": "calc(100% - 40px)",
      }
    }
  >
    Email
  </label>
  <div
    className="c2"
  >
    <div
      className="c3 c4"
      data-testid="icon-field-prefix-icons"
    >
      <svg
        aria-hidden={true}
        className="c5 c6"
        fill="currentcolor"
        focusable={false}
        height={24}
        role="img"
        tabIndex={-1}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
        />
      </svg>
    </div>
    <input
      className="c7"
      fontSize={
        Array [
          2,
          null,
          1,
        ]
      }
      id="email"
      name="email"
      size="lg"
      style={
        Object {
          "paddingBottom": undefined,
          "paddingTop": undefined,
          "transitionDuration": ".1s",
          "transitionProperty": "padding-top, padding-bottom",
        }
      }
    />
  </div>
</div>
`;

exports[`FormField renders with Select  1`] = `
.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  color: #4f6f8f;
  margin-left: -32px;
  width: 24px;
}

.c6 {
  outline: none;
}

.c1 {
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: #4f6f8f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 10px;
  margin-left: 12px;
  padding-top: 6px;
  margin-bottom: -20px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c0 {
  border-radius: 12px;
}

.c3 {
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  pointer-events: none;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  border-style: solid;
  border-width: 1px;
  border-radius: 12px;
  color: inherit;
  display: block;
  font-family: inherit;
  text-overflow: ellipsis;
  width: 100%;
  padding: 14px 32px 14px 12px;
  border-color: #c0cad5;
  font-size: 16px;
  margin: 0px;
}

.c4::-ms-expand {
  display: none;
}

.c4:hover {
  cursor: pointer;
}

.c4:disabled {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
  opacity: 1;
}

.c4:disabled ~ svg {
  color: #4f6f8f;
}

.c4:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c4 {
    font-size: 14px;
  }
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="text.light"
    fontSize={10}
    fontWeight={700}
    letterSpacing="0.2px"
    style={
      Object {
        "height": 20,
        "opacity": 1,
        "pointerEvents": "none",
        "position": "relative",
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
        "width": "calc(100% - 12px)",
      }
    }
  >
    Pick Option
  </label>
  <div
    className="c2"
  >
    <div
      className="c3 c2"
      width={1}
    >
      <select
        className="c4"
        fontSize={
          Array [
            2,
            null,
            1,
          ]
        }
        size="lg"
        style={
          Object {
            "paddingBottom": "8px",
            "paddingTop": "20px",
            "transitionDuration": ".1s",
            "transitionProperty": "padding-top, padding-bottom",
          }
        }
      />
      <svg
        aria-hidden={true}
        className="c5 c6 c7"
        color="text.light"
        fill="currentcolor"
        focusable={false}
        height={24}
        role="img"
        tabIndex={-1}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`FormField renders with Select and Icon 1`] = `
.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-left: 8px;
  margin-right: 8px;
  width: 24px;
}

.c9 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  color: #4f6f8f;
  margin-left: -32px;
  width: 24px;
}

.c10 {
  outline: none;
}

.c6 {
  outline: none;
}

.c1 {
  display: block;
  width: 100%;
  margin: 0;
  z-index: 1;
  color: #4f6f8f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 10px;
  margin-left: 40px;
  padding-top: 6px;
  margin-bottom: -20px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c0 {
  border-radius: 12px;
}

.c3 {
  margin-right: -40px;
}

.c7 {
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  pointer-events: none;
}

.c8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  border-style: solid;
  border-width: 1px;
  border-radius: 12px;
  color: inherit;
  display: block;
  font-family: inherit;
  text-overflow: ellipsis;
  width: 100%;
  padding: 14px 32px 14px 12px;
  border-color: #c0cad5;
  padding-left: 40px;
  font-size: 16px;
  margin: 0px;
}

.c8::-ms-expand {
  display: none;
}

.c8:hover {
  cursor: pointer;
}

.c8:disabled {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
  opacity: 1;
}

.c8:disabled ~ svg {
  color: #4f6f8f;
}

.c8:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c8 {
    font-size: 14px;
  }
}

<div
  className="c0 "
>
  <label
    className="c1"
    color="text.light"
    fontSize={10}
    fontWeight={700}
    letterSpacing="0.2px"
    style={
      Object {
        "height": 20,
        "opacity": 1,
        "pointerEvents": "none",
        "position": "relative",
        "transitionDuration": ".1s",
        "transitionProperty": "opacity",
        "width": "calc(100% - 40px)",
      }
    }
  >
    Pick Email Address
  </label>
  <div
    className="c2"
  >
    <div
      className="c3 c4"
      data-testid="icon-field-prefix-icons"
    >
      <svg
        aria-hidden={true}
        className="c5 c6"
        fill="currentcolor"
        focusable={false}
        height={24}
        role="img"
        tabIndex={-1}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
        />
      </svg>
    </div>
    <div
      className="c7 c2"
      width={1}
    >
      <select
        className="c8"
        fontSize={
          Array [
            2,
            null,
            1,
          ]
        }
        size="lg"
        style={
          Object {
            "paddingBottom": "8px",
            "paddingTop": "20px",
            "transitionDuration": ".1s",
            "transitionProperty": "padding-top, padding-bottom",
          }
        }
      />
      <svg
        aria-hidden={true}
        className="c9 c10 c11"
        color="text.light"
        fill="currentcolor"
        focusable={false}
        height={24}
        role="img"
        tabIndex={-1}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;
